### PR TITLE
Fix #236: Prevent room deletion if there are issues associated with it

### LINKED ISSUE #146: Bug(Room): `odel` should either delete all issues associated or restrict the deletion

### LINKED ISSUE #224: [PE-D] Removing a room should remove its related issues
Expected:
Deleting a room should have its associated issues removed as well.

Actual:
The associated issues remain without any indication that the room has been deleted.

Maybe consider deleting those issues too or keep them as a record but indicate that its associated room has been deleted.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Tomashiwa/ped#8

### LINKED ISSUE #236: [PE-D] Olist not sorting by room number
00-200 is less than all room numbers above it

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: BenedictBCJJ/ped#10

### PR DESCRIPTION
**Pull request by DrWala**
### What this does
Closes #224 by blocking usage of odel unless all associated issues with it are deleted
- Fixes #146 

### How to test
1. Add a bunch of issues to a room
2. Try to delete the room. It should fail
3. Try to delete a room without issues. It should not fail
4. Do anything else to try and break this

### CODE REVIEW

File: src/main/java/seedu/address/model/issue/IssueList.java [hunk 1]
```diff
@@ -10,10 +10,11 @@
 import javafx.collections.ObservableList;
 import seedu.address.model.issue.exceptions.DuplicateIssueException;
 import seedu.address.model.issue.exceptions.IssueNotFoundException;
+import seedu.address.model.room.Room;
 
 /**
  * A list of issues that does not allow nulls.
- *
+ * <p>
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/model/issue/IssueList.java:17]: What's this tag for?

File: src/main/java/seedu/address/model/issue/IssueList.java [hunk 2]
```diff
@@ -99,6 +100,17 @@ public void setIssues(List<Issue> issues) {
         FXCollections.sort(internalList);
     }
 
+    /**
+     * Checks if any issues have the given room associated with it
+     *
+     * @param target Room to check if it has issues associated with it.
+     * @return true if there are issues with the given room associated with it
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/model/issue/IssueList.java:107]: Is this missing a fullstop, and should it start with a capital letter?

Saw this in `Model.java` and `AddressBook.java` too.

File: src/main/java/seedu/address/model/issue/IssueList.java [hunk 3]
```diff
@@ -99,6 +100,17 @@ public void setIssues(List<Issue> issues) {
         FXCollections.sort(internalList);
     }
 
+    /**
+     * Checks if any issues have the given room associated with it
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/model/issue/IssueList.java:104]: Is this missing a fullstop?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -45,7 +45,7 @@
      * Initializes a ModelManager with the given addressBook, userPrefs and commandHistory.
      */
     public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs,
-            ReadOnlyCommandHistory commandHistory) {
+                        ReadOnlyCommandHistory commandHistory) {
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/model/ModelManager.java:48]: Is the spacing here supposed to be 8 spaces instead of 16?

**Review by benedictkhoomw — CHANGES_REQUESTED** [review]
Cloned and smoke tested. Looks good, apart from a few nits.

File: src/main/java/seedu/address/model/issue/IssueList.java (hunk 1 above)
Comment (DrWala) [reply src/main/java/seedu/address/model/issue/IssueList.java:17]: It is auto-added by Intellij as part of the normal way of providing paragraph breaks in javadocs. It is conformant to the javadoc standards: https://www.oracle.com/sg/technical-resources/articles/java/javadoc-tool.html

File: src/main/java/seedu/address/model/ModelManager.java (hunk 4 above)
Comment (DrWala) [reply src/main/java/seedu/address/model/ModelManager.java:48]: This was automatically done by the code formatter, and checkstyle seems ok with it. It seems to follow the "conventional indentation" guidelines set out by oracle: https://www.oracle.com/java/technologies/javase/codeconventions-indentation.html

However, the textbook (https://se-education.org/guides/conventions/java/intermediate.html#layout) says line breaks should only double the indentation. Let me know what you want to do about this.

File: src/main/java/seedu/address/model/issue/IssueList.java (hunk 3 above)
Comment (DrWala) [reply src/main/java/seedu/address/model/issue/IssueList.java:104]: Yes.

File: src/main/java/seedu/address/model/issue/IssueList.java (hunk 2 above)
Comment (DrWala) [reply src/main/java/seedu/address/model/issue/IssueList.java:107]: Yes, fixed.

**Review by benedictkhoomw — APPROVED** [review]
LGTM.

**Pull request merged**